profitbricks/src/main/java/org/jclouds/profitbricks/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,269 @@
+      Hardware hardware = template.getHardware();
+
+      waitDcUntilAvailable.apply(dataCenterId);
+
+      TemplateOptions options = template.getOptions();
+      String password = options.hasLoginPassword() ? options.getLoginPassword() : Passwords.generate();
+
+      List<? extends Volume> volumes = hardware.getVolumes();
+      List<String> storageIds = Lists.newArrayListWithExpectedSize(volumes.size());
+      int i = 1;
+      for (Volume volume : volumes) {
+         logger.trace("<< provisioning storage '%s'", volume);
+         String id = api.storageApi().createStorage(
+                 Storage.Request.creatingBuilder()
+                 .dataCenterId(dataCenterId)
+                 // put image to first storage
+                 .mountImageId(i == 1 ? template.getImage().getId() : "")
+                 .imagePassword(password)
+                 .name(format("%s-disk-%d", name, i++))
+                 .size(volume.getSize())
+                 .build());
+         waitDcUntilAvailable.apply(dataCenterId);
+         storageIds.add(id);
+         logger.trace(">> provisioning complete for storage. returned id='%s'", id);
+      }
+
+      Double cores = 0d;
+      for (Processor processor : hardware.getProcessors())
+         cores += processor.getCores();
+
+      String storageBootDeviceId = Iterables.getFirst(storageIds, "");
+      Server.Request.CreatePayload serverRequest = Server.Request.creatingBuilder()
+              .dataCenterId(dataCenterId)
+              .name(name)
+              .bootFromStorageId(storageBootDeviceId)
+              .cores(cores.intValue())
+              .ram(hardware.getRam())
+              .availabilityZone(AvailabilityZone.AUTO)
+              .hasInternetAccess(true)
+              .build();
+
+      logger.trace("<< provisioning server '%s'", serverRequest);
+      String serverId = api.serverApi().createServer(serverRequest);
+      waitDcUntilAvailable.apply(dataCenterId);
+      logger.trace(">> provisioning complete for server. returned id='%s'", serverId);
+
+      final int storageCount = storageIds.size();
+      for (int j = 1; j < storageCount; j++) { // skip first; already connected
+         String storageId = storageIds.get(j);
+         logger.trace("<< connecting storage '%s' to server '%s'", storageId, serverId);
+         api.storageApi().connectStorageToServer(
+                 Storage.Request.connectingBuilder()
+                 .storageId(storageId)
+                 .serverId(serverId)
+                 .build());
+
+         waitDcUntilAvailable.apply(dataCenterId);
+         logger.trace(">> storage connected.");
+      }
+
+      Server server = getNode(serverId);
+
+      LoginCredentials serverCredentials = LoginCredentials.builder()
+              .user("root") // always root user
+              .password(password)
+              .build();
+
+      return new NodeAndInitialCredentials<Server>(server, serverId, serverCredentials);
+   }
```

Comment:
This method creates several resources in the provider. If it fails at some point, we should take care of rolling back the created stuff (at least we should try), in order to avoid leaking resources.